### PR TITLE
Namespace exemption for kube-system azure-arc

### PR DESCRIPTION
**Pull request by nshankar13**
Added openservicemesh.io/ignore label in the kube-system and azure-arc namespaces.

### CODE REVIEW

File: charts/osm/templates/azure-policy-osm-label.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# 
+# The job in this YAML file tries to add 'openservicemesh.io/ignore' label to 'kube-system' and 'azure-arc' namespaces.
```
Comment (snehachhabria) [inline charts/osm/templates/azure-policy-osm-label.yml:2]: Should `arc-osm-system` also be included in the list of namespaces having this label ?

Comment (ritazh) [reply charts/osm/templates/azure-policy-osm-label.yml:2]: This pre-install hook will only take care of existing namespaces in the cluster before deploying the rest of the resources in this chart. To ensure the namespace containing osm resources (ie. `arc-osm-system`) has the exclusion labels, that should be taken care of by 1 and 2 in this [comment](https://github.com/openservicemesh/osm/issues/1817#issuecomment-707853626).

**Comment by ritazh** [comment]
@nshankar13 can you please rebase?

File: charts/osm/templates/azure-policy-osm-label.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# The job is executed in helm pre-install phase of helm life cycle.
+#
+# For namespaces with 'openserviemesh.io/ignore' label...
+#     1. Kubernetes admission requests with violations are not denied by Gatekeeper.
+#     2. Auditing results are still available.
+#
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: azure-policy-osm-label-account
```
Comment (ritazh) [inline charts/osm/templates/azure-policy-osm-label.yml:12]: you can remove all mention of "azure-policy"

File: charts/osm/templates/azure-policy-osm-label.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  namespace: kube-system
```
Comment (ritazh) [inline charts/osm/templates/azure-policy-osm-label.yml:13]: you can get this namespace from values.yaml once Sneha adds it in the other PR

File: scripts/osmextension.json [hunk 4]
```diff
@@ -1,12 +0,0 @@
-{
-  "properties": {
-    "extensionType": "Microsoft.openservicemesh",
```
Comment (ritazh) [inline scripts/osmextension.json:3]: why is this file deleted?

Comment (nshankar13) [reply scripts/osmextension.json:3]: Accident

File: charts/osm/templates/azure-policy-osm-label.yml (hunk 3 above)
Comment (nshankar13) [reply charts/osm/templates/azure-policy-osm-label.yml:13]: Does the "arc-osm-system" namespace that Sneha is adding impact anything in the osm-label job? Because regardless we are still adding the openservicemesh.io/ignore label only to the azure-arc and kube-system namespaces?

File: test/bats/test.bats [hunk 5]
```diff
@@ -19,3 +19,8 @@ SLEEP_TIME=1
     run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get mutatingwebhookconfigurations.admissionregistration.k8s.io osm-webhook-osm"
     assert_success
 }
+
+@test "osm-label job in kube-system has successfully completed" {
+    run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl wait --for=condition=complete job osm-label --namespace kube-system --timeout=60s"
```
Comment (ritazh) [inline test/bats/test.bats:24]: can you add a test that checks for the new label no kube-system?

**Comment by nshankar13** [comment]
@ritazh ready for review

File: scripts/osmextension.json [hunk 6]
```diff
@@ -9,4 +9,4 @@
       }
     }
   }
-}
+}
```
Comment (ritazh) [inline scripts/osmextension.json:12]: nit: new line

File: scripts/add-extension.sh [hunk 7]
```diff
@@ -3,18 +3,26 @@
...
 az resource tag --tags logAnalyticsWorkspaceResourceId=/$RESOURCEID --ids /$RESOURCEID
+
 # get extensions enabled on this cluster
 az rest --method GET --uri "https://management.azure.com/$RESOURCEID/providers/Microsoft.KubernetesConfiguration/extensions?api-Version=2020-07-01-preview"
+
 # enable the osm extension
-az rest --method PUT --uri "https://management.azure.com/$RESOURCEID/providers/Microsoft.KubernetesConfiguration/extensions/osm?api-Version=2020-07-01-preview" --body @osmextension.json --debug
-# wait few mins to see the osm resources created
-helm ls --all --all-namespaces                                                                        
+az rest --method PUT --uri "https://management.azure.com/$RESOURCEID/providers/Microsoft.KubernetesConfiguration/extensions/osm?api-Version=2020-07-01-preview" --body @osm_extension.json --debug
```
Comment (ritazh) [inline scripts/add-extension.sh:28]: should we also update the content of `osm_extension.json`?

File: Makefile [hunk 8]
```diff
@@ -14,6 +14,8 @@ e2e-bootstrap:
 	if [ $$(kind get clusters) ]; then kind delete cluster; fi
 	# Create a new kind cluster
 	TERM=dumb kind create cluster --image kindest/node:${KUBERNETES_VERSION}
+	# Create namespaces to test openservicemesh.io/ignore label
+	kubectl create namespace azure-arc
```
Comment (ritazh) [inline Makefile:18]: Should we create the namespaces in the same place in e2e-helm-deploy?

File: test/bats/test.bats [hunk 9]
```diff
@@ -19,3 +19,21 @@ SLEEP_TIME=1
     run wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get mutatingwebhookconfigurations.admissionregistration.k8s.io arc-osm-webhook-osm"
     assert_success
 }
+
+@test "openservicemesh.io/ignore is true in kube-system" { 
+    kubectl get namespace kube-system -o json > test.json
```
Comment (ritazh) [inline test/bats/test.bats:24]:
```suggestion
    test_label=$(kubectl get namespace kube-system -o json | jq '.metadata.labels["openservicemesh.io/ignore"]')
    run wait_for_process $WAIT_TIME $SLEEP_TIME test_label
```

If you use pipe then you dont need to rm the test.json file.

File: osm_extension.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
+{
+  "properties": {
```
Comment (ritazh) [inline osm_extension.json:2]: you don't need to check this in. add to gitignore?

File: test/bats/helpers.bash [hunk 11]
```diff
@@ -9,6 +9,15 @@ assert_success() {
   fi
 }
 
+assert_output_true() {
```
Comment (ritazh) [inline test/bats/helpers.bash:12]: this is no long used

File: charts/osm/values.yaml [hunk 12]
```diff
@@ -18,6 +18,7 @@ OpenServiceMesh:
   useHTTPSIngress: false
   envoyLogLevel: error
   namespace: arc-osm-system
+  namespaces: "kube-system hello-there azure-arc arc-osm-system"
```
Comment (ritazh) [inline charts/osm/values.yaml:21]: remove hello-there

File: charts/osm/templates/osm-label.yml [hunk 13]
```diff
@@ -0,0 +1,144 @@
+# 
+# The job in this YAML file tries to add 'openservicemesh.io/ignore' label to 'kube-system' and 'azure-arc' namespaces.
+# The job is executed in helm pre-install phase of helm life cycle.
+#
+# For namespaces with 'openserviemesh.io/ignore' label...
+#     1. Kubernetes admission requests with violations are not denied by Gatekeeper.
```
Comment (ritazh) [inline charts/osm/templates/osm-label.yml:6]: pls update this comment for osm

File: charts/osm/templates/osm-label.yml [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+#     2. Auditing results are still available.
```
Comment (ritazh) [inline charts/osm/templates/osm-label.yml:7]: remove

**Review by ritazh — APPROVED** [review]
lgtm

File: charts/osm/values.yaml [hunk 15]
```diff
@@ -18,6 +18,7 @@ OpenServiceMesh:
   useHTTPSIngress: false
   envoyLogLevel: error
   namespace: arc-osm-system
+  namespaces: "kube-system azure-arc arc-osm-system"
```
Comment (snehachhabria) [inline charts/osm/values.yaml:21]: I feel have a value namespace and namespaces is little confusing and overloads the namespace term. If this is used to ignore namespaces, could we rename it to `ignoreNamespaces` ?

Comment (ritazh) [reply charts/osm/values.yaml:21]: +1

**Review by snehachhabria — APPROVED** [review]
